src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/IProvenBlockHeaderRepository.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Interface for database <see cref="ProvenBlockHeader"></see> repository.
+    /// </summary>
+    public interface IProvenBlockHeaderRepository : IDisposable
+    {
+        /// <summary>Loads <see cref="ProvenBlockHeader"/> items from the database.</summary>
+        /// <param name="blockHash">BlockId to initial the database with.</param>
+        /// <param name="cancellationToken">Cancellation that allows aborting the operation at any stage.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task InitializeAsync(uint256 blockHash = null, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
cancellation is in every interface but not used anywhere. Also methods that support cancellation are not invoked with non-default cancellation in the methods you've introduced for some reason.
I think it will be cleaner to get rid of it cancellation at all.

Reply:
The `cancellationToken` is available in case the calling code wants to send on its token.  

I guess `ProvenBlockHeaderStore` has a handle on `nodeLifetime.ApplicationStopping`.  So it can decide whether to call into the `ProvenBlockHeaderRepository`.

I've removed the parameter and it does look a lot cleaner.